vaxrank/report.py

```diff
@@ -76,37 +86,101 @@ def ascii_report_from_ranked_vaccine_peptides(
-                "\t\t   - Reads fully spanning cDNA sequence(s): %d" % (
-                    mutant_protein_fragment.n_alt_reads_supporting_protein_sequence))
-            lines.append(
-                "\t\t   - Mutant amino acids: %d" % (
-                    mutant_protein_fragment.n_mutant_amino_acids))
-            lines.append(
-                "\t\t   - Mutation distance from edge: %d" % (
-                    mutant_protein_fragment.mutation_distance_from_edge))
-            lines.append("\t\t   - Predicted mutant epitopes:")
+            epitopes = []
+            peptide_dict = {
+                'num': roman.toRoman(j + 1).lower(),
+                'aa_before_mutation': aa_before_mutation,
+                'aa_mutant': aa_mutant,
+                'aa_after_mutation': aa_after_mutation,
+                'score': vaccine_peptide.combined_score,
+                'length': len(amino_acids),
+                'expression_score': vaccine_peptide.expression_score,
+                'mutant_epitope_score': vaccine_peptide.mutant_epitope_score,
+                'wildtype_epitope_score': vaccine_peptide.wildtype_epitope_score,
+                'num_alt_reads_supporting_protein_sequence':
+                    mutant_protein_fragment.n_alt_reads_supporting_protein_sequence,
+                'num_mutant_amino_acids':
+                    mutant_protein_fragment.n_mutant_amino_acids,
+                'mutation_distance_from_edge':
+                    mutant_protein_fragment.mutation_distance_from_edge,
+                'epitopes': epitopes,
+            }
+            
+            # compile epitope info
             for epitope_prediction in vaccine_peptide.epitope_predictions:
                 if epitope_prediction.overlaps_mutation and epitope_prediction.ic50 <= 2000:
-                    lines.append("\t\t\t * %s %0.2f (%s)" % (
-                        epitope_prediction.peptide_sequence,
-                        epitope_prediction.ic50,
-                        epitope_prediction.allele))
-    return "\n".join(lines)
+                    epitope_dict = {
+                        'sequence': epitope_prediction.peptide_sequence,
+                        'ic50': epitope_prediction.ic50,
+                        'allele': epitope_prediction.allele,
+                    }
+                    epitopes.append(epitope_dict)
+
+            peptides.append(peptide_dict)
+        variants.append(variant_dict)
+
+    template_data['variants'] = variants
+    return template_data
+
+
+def make_ascii_report_from_ranked_vaccine_peptides(
+        ranked_variants_with_vaccine_peptides,
+        mhc_alleles,
+        variants,
+        bam_path,
+        ascii_report_path):
+    template = JINJA_ENVIRONMENT.get_template('templates/template.txt')
+    template_data = compute_template_data(
+        ranked_variants_with_vaccine_peptides,
+        mhc_alleles,
+        variants,
+        bam_path)
+    report = template.render(template_data)
+    print(report)
```

Comment:
I think we probably shouldn't print the report, though I know we were doing this before